code/processes/housekeeping.q

```diff
@@ -96,20 +96,25 @@ zip:{[FILE]
 
 //-locates files with path, matching string and age
 find:{[path;match;age;agemin]
+	//returns empty list for files if match is an empty string
+        if[""~match;
+          .lg.o[`housekeeping;"No matching files located"];
+          :();
+        ];
 	//renames the path to a windows readable format
 	PATH:ssr[path;"/";"\\"];
-	//searches for files and refines return to usable format
+	//error and info for find function 
+	err:{.lg.e[`housekeeping;"Find function failed: ", x]; ()};
+        //searches for files and refines return to usable format
 	$[0=agemin;
 	files:.[{[PATH;match;age].lg.o[`housekeeping;"Searching for: ", match];
-		system "z 1";fulllist:-5_(5_system "dir ",PATH,match, " /s");
+		system "z 1";fulllist:-2_(5_system "dir ",PATH,match);
```

Comment:
So I think the crux of this problem is that the functionality of these system commands differs between windows versions (windows 7, windows 10, windows server 20XX etc.). I believe we've changed this exact section before. So what we need to do is extract this functionality into a separate function, which we can extend based on the OS. 

If this bit is what we need now: 

-2_(5_system "dir ",PATH,match)

then that should be the option for whatever version of windows this works for. I'm pretty sure this would've worked in the past too: 
-5_(5_system "dir ",PATH,match, " /s")

So can you create a separate function to encapsulate this and have an OS switch e.g. 

winfind:{$[.windows.version=`w10; .... ; .windows.version=`w7; ... } 

etc. We will have to pick an appropriate default value. 

This should make it more extendible in the future. For now .windows.version will just have to be manually set by the user in config I think unless we can work out a way of working out the version from the cmd prompt (though this itself my then be dependent on the version :-S... thanks Windows!).  kdb will only tell us w32 or w64 which isn't enough

Reply:
Fair enough, will need this documented. `-2_5_` and no `/s` is what I needed on Windows 8 FWIW.

Reply:
Windows 7 is the same as Windows 8, with `/s` meaning search subdirs.